server.js

```diff
@@ -14,6 +14,48 @@ app.get("/", function (request, response) {
 });
 
 // TODO add your routes and helper functions here
+//read all bookings
+app.get("/bookings", (req, res) => {
+  res.send(bookings);
+});
+app.post("/bookings", (req, res) => {
+  //validate our input
+  const newBooking = req.body;
+  newBooking.id = bookings.length + 1;
+  //build our booking
+  for (let field in newBooking) {
+    if (!newBooking[field]) {
```

Comment:
This `if` statement is not correct as it will **always** evaluate to **false**.

One approach you can try is declaring an array containing all the expected fields and looping through it to check if all of them exist in `newBooking`.